test/test_torch.py

```diff
@@ -5260,6 +5260,22 @@ def test_multiplication_numpy_scalar(self):
                 self.assertTrue(r2.dtype == t_dtype)
                 self.assertTrue(r2.requires_grad)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_parse_numpy_int(self):
+        # https://github.com/pytorch/pytorch/issues/29252
+        for nptype in [np.int16, np.int8, np.uint8, np.int32, np.int64]:
+            np_arr = np.array([5], dtype=nptype)
+            np_val = np_arr[0]
+            self.assertEqual(torch.Storage(np_val).size(), 5)
+
+            tensor = torch.tensor([2], dtype=torch.int)
+            tensor[0] = np_val
+            self.assertEqual(tensor[0], np_val)
+
+            t = torch.from_numpy(np_arr)
+            self.assertEqual((t + np_val).dtype, (t + t[0].item()).dtype)
```

Comment:
Isn't `(t + t[0].item()).dtype` just `t.dtype` ? 

What exactly is this testing?

Reply:
ah, yeah this isn't necessary and I'll fix it. t[0].item() will always be long or double, so it is "special" in a way, but within a type category it won't ever actually change the type of the result since it's being added to a tensor with dims.

Reply:
fixed